### PR TITLE
Exit when invalid engine is selected

### LINKED ISSUE #4584: [0.9.4.2] Storing strings is not working
Hi,

I am trying to store strings in influxdb 0.9.4.2, using influxdb-java. Line protocol output of influxdb-java looks like: ```test,identifier=Testperson value="test test test" 1445944679577000000```

But no matter if I am using influxdb-java or the influxdb web interface (with the same line protocol data), the point is not stored to my database. Web interface doesn't report any error, but I can't query data by
```SELECT value FROM test```, it says ```Success! (no results to display)``` then.

When I change my timestamp to seconds instead of nanoseconds, everything is working fine. Also if I am storing floats instead of strings everything is fine. Am I doing something wrong or is this an error in influxdb?

Greetings
Dennis

### LINKED ISSUE #4585: [feature request] output meaningful error if the configuration file has an unrecognized engine type

### PR DESCRIPTION
**Pull request by otoolep**
Fix #4585, related to #4583

### CODE REVIEW

**Comment by otoolep** [comment]
It is not possible to exit short of a panic, but this change makes it very clear in the logs.

```
[write] 2015/10/27 11:56:14 write failed for shard 1 on node 1: new engine: engine xxx does not exist
[monitor] 2015/10/27 11:56:14 failed to store statistics: write failed: new engine: engine xxx does not exist
[write] 2015/10/27 11:56:24 write failed for shard 1 on node 1: new engine: engine xxx does not exist
[monitor] 2015/10/27 11:56:24 failed to store statistics: write failed: new engine: engine xxx does not exist
```

**Comment by otoolep** [comment]
@jwilder @beckettsean

**Comment by beckettsean** [comment]
works for me

**Comment by jwilder** [comment]
Could we validate this in the config?  i.e.: https://github.com/influxdb/influxdb/blob/master/cmd/influxd/run/command.go#L104

**Comment by otoolep** [comment]
Yeah, that might be better. Let me try that.

**Comment by otoolep** [comment]
OK, updated as per @jwilder . Program now exits, which is better.
```
$ ./influxd -config ~/Dropbox/influxdb/config.graphite.toml 

 8888888           .d888 888                   8888888b.  888888b.
   888            d88P"  888                   888  "Y88b 888  "88b
   888            888    888                   888    888 888  .88P
   888   88888b.  888888 888 888  888 888  888 888    888 8888888K.
   888   888 "88b 888    888 888  888  Y8bd8P' 888    888 888  "Y88b
   888   888  888 888    888 888  888   X88K   888    888 888    888
   888   888  888 888    888 Y88b 888 .d8""8b. 888  .d88P 888   d88P
 8888888 888  888 888    888  "Y88888 888  888 8888888P"  8888888P"

2015/10/27 17:10:46 InfluxDB starting, version 0.9, branch unknown, commit unknown, built unknown
2015/10/27 17:10:46 Go version go1.5.1, GOMAXPROCS set to 8
2015/10/27 17:10:46 Using configuration at: /home/philip/Dropbox/influxdb/config.graphite.toml
run: unrecognized engine xxx. To generate a valid configuration file run `influxd config > influxdb.generated.conf`.
```

File: tsdb/config.go [hunk 1]
```diff
@@ -128,3 +130,24 @@ func NewConfig() Config {
 		QueryLogEnabled: true,
 	}
 }
+
+func (c *Config) Validate() error {
```
Comment (otoolep) [inline tsdb/config.go]: This pattern, of adding `Validate` to individual config types, exists elsewhere in the code.

**Comment by jwilder** [comment]
:+1:

**Pull request merged**